src/Symfony/Bundle/FrameworkBundle/Logger/Logger.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Logger;
+
+use Psr\Log\AbstractLogger;
+use Psr\Log\InvalidArgumentException;
+use Psr\Log\LogLevel;
+
+/**
+ * Minimalist PSR-3 logger designed to write in stdout, stderr or any other stream.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+final class Logger extends AbstractLogger
+{
+    private static $defaultOutputByLevel = array(
+        LogLevel::EMERGENCY => 'php://stderr',
+        LogLevel::ALERT => 'php://stderr',
+        LogLevel::CRITICAL => 'php://stderr',
+        LogLevel::ERROR => 'php://stderr',
+        LogLevel::WARNING => 'php://stderr',
+        LogLevel::NOTICE => false,
+        LogLevel::INFO => false,
+        LogLevel::DEBUG => false,
+    );
+    private $outputByLevel;
+    private $format;
+    private $dateFormat;
+    private $errored = false;
+    private $handles = array();
+
+    public function __construct(array $outputByLevel = array(), $format = '[%s] %s', $dateFormat = \DateTime::RFC3339)
+    {
+        $this->outputByLevel = $outputByLevel + self::$defaultOutputByLevel;
+        $this->format = $format;
+        $this->dateFormat = $dateFormat;
+    }
+
+    public function __destruct()
+    {
+        foreach ($this->handles as $handle) {
+            \fclose($handle);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function log($level, $message, array $context = array())
+    {
+        if (!isset($this->outputByLevel[$level])) {
+            throw new InvalidArgumentException(\sprintf('The log level "%s" does not exist.', $level));
+        }
+
+        if (!$this->errored) {
+            $this->errored = \in_array($level, array(LogLevel::EMERGENCY, LogLevel::ALERT, LogLevel::CRITICAL, LogLevel::ERROR), true);
+        }
+
+        if ($stream = $this->outputByLevel[$level]) {
+            if (!isset($this->handles[$stream])) {
+                $this->handles[$stream] = \fopen($stream, 'a');
+            }
+
+            \fwrite($this->handles[$stream], \sprintf($this->format, $level, $this->interpolate($message, $context)).PHP_EOL);
```

Comment:
`PHP_EOL` needs a leading `\` too

Reply:
we don't want to root-namespace everything, see https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/3048 for some background.

Reply:
Well, for constants, it would make sense though (as they are resolved at compile time for internal constants)